cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -95,6 +116,19 @@ func doSystemdMountImpl(what, where string, opts *systemdMountOptions) error {
 	whereEscaped := systemd.EscapeUnitNamePath(where)
 	unitName := whereEscaped + ".mount"
 
+	if opts.Tmpfs && what == "" {
+		what = "tmpfs"
+	}
+
+	forbiddenChars := `\,:" `
+	if pathContainsAny(what, forbiddenChars) {
+		return fmt.Errorf("cannot mount %q at %q: what systemd-mount argument contains forbidden characters. %q contains one of \"%s\".", what, where, what, forbiddenChars)
+	}
+
+	if pathContainsAny(where, forbiddenChars) {
+		return fmt.Errorf("cannot mount %q at %q: where systemd-mount argument contains forbidden characters. %q contains one of \"%s\".", what, where, where, forbiddenChars)
+	}
```

Comment:
I do not think we need this. Those paths are not going into the options string.

Reply:
I removed the restrictions from the what and where arguments.